### ISSUE TITLE
Sniff doesn't work on getting nodes_to_host - in EL 2.0rc1 : http_address changed 

### ISSUE THREAD

**Issue by easyrasta** [elastic/elasticsearch-js#281]
I added snifInterval and ELjs throw me an exception on apply regex on host field
```
Error: node's http_address property ("192.168.0.33:9200") does not match the expected pattern /\[\/*([^:]+):(\d+)\]/.
/..../node_modules/@.../elasticdb/node_modules/elasticsearch/src/lib/nodes_to_host.js:13:15
```
Regex is invalid with this EL version
It should be:
```
var extractHostPartsRE = /([^:]+):(\d+)/;
```

**Comment by royrusso** [comment]
Same issue here on ES 2.0.0-rc1 using the jquery method:


```
 Error: Unable to convert node list from undefined to hosts durring sniff. Encountered error:
  Error: node's http_address property ("127.0.0.1:9200") does not match the expected pattern /\[\/*([^:]+):(\d+)\]/.
      at http://localhost:63343/elasticsearch-HQ/js/lib/elasticsearch/elasticsearch.jquery.js:33373:15
      at http://localhost:63343/elasticsearch-HQ/js/lib/elasticsearch/elasticsearch.jquery.js:16436:16
      at http://localhost:63343/elasticsearch-HQ/js/lib/elasticsearch/elasticsearch.jquery.js:9418:15
```

I confirmed that @easyrasta regexp fix works with this version, but can't test for backwards compatibility.

**Comment by glivolsi** [comment]
Same issue "easyrasta" regexp fix it, please resolve...!!

**Comment by easyrasta** [comment]
@splager May be close by 7319b49

**Comment by spalger** [comment]
Correct, this is fixed in 9.0.1 which was published yesterday.

**Comment by agencia** [comment]
Not sure if this is a regression bug but i got the same on __elasticsearch@10.0.1__

```
Elasticsearch ERROR: 2015-12-13T05:18:13Z
  Error: Unable to convert node list from undefined to hosts durring sniff. Encountered error:
  Error: expected node's http_address property ("monitor.agei.mx/10.0.1.43:9200") to match either /^([\.:0-9a-f]*):([0-9]+)?$/ or /\[\/*([^:]+):(\d+)\]/.
```
